trin-core/src/portalnet/overlay.rs

```diff
@@ -801,22 +804,36 @@ where
 }
 
 /// Randomly select log2 nodes. log2() is stable only for floats, that's why we cast it first to f32
-fn log2_random_enrs(enrs: Vec<Enr>) -> anyhow::Result<Vec<Enr>> {
-    if enrs.is_empty() {
-        return Err(anyhow!("Expected non-empty vector of ENRs"));
-    }
-    // log2(1) is zero but we want to propagate to at least one node
-    if enrs.len() == 1 {
-        return Ok(enrs);
-    }
-
-    // Greedy gossip by ceiling the log2 value
-    let log2_value = (enrs.len() as f32).log2().ceil();
-
+fn random_enrs(num_enrs: usize, enrs: Vec<Enr>) -> Vec<Enr> {
```

Comment:
Maybe rename this to something like: 
```suggestion
fn select_random_enrs(num_enrs: usize, enrs: Vec<Enr>) -> Vec<Enr> {
```